src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,99 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int ZERO = 0;
+    private static final int MIN_AGE = 18;
+    private static final int MAX_AGE = 120;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUser(user);
+        checkAgeNull(user);
+        checkAgeLess18(user);
+        checkAgeBigger100(user);
+        checkAgeZero(user);
+        checkLoginNull(user);
+        checkLoginIncorrect(user);
+        checkPasswordNull(user);
+        checkPasswordLess8(user);
+        checkIdNull(user);
+        checkIdNegative(user);
+        checkStorageContainsUserAlready(user);
+        storageDao.add(user);
+        return user;
+    }
+
+    private void checkStorageContainsUserAlready(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new IncorrectDataExeption("User already existed in storage");
+        }
+    }
+
+    private void checkUser(User user) {
+        if (user == null) {
+            throw new IncorrectDataExeption("User can not be null");
+        }
+    }
+
+    private void checkAgeZero(User user) {
+        if (user.getAge() < ZERO) {
+            throw new IncorrectDataExeption("Age can not be less than 0");
+        }
+    }
+
+    private void checkAgeNull(User user) {
+        if (user.getAge() == null) {
+            throw new IncorrectDataExeption("Age can not be null");
+        }
+    }
+
+    private void checkAgeLess18(User user) {
+        if (user.getAge() < MIN_AGE) {
+            throw new IncorrectDataExeption("Age can not be less than 18");
+        }
+    }
+
+    private void checkAgeBigger100(User user) {
```

Comment:
change age in method name

Reply:
Do you check ages bigger than 100 or 120?